adapters/mock-da/src/service.rs

```diff
@@ -122,6 +128,9 @@ impl DaService for MockDaService {
     /// It is possible to read non-finalized and last finalized blocks multiple times
     /// Finalized blocks must be read in order.
     async fn get_block_at(&self, height: u64) -> Result<Self::FilteredBlock, Self::Error> {
+        if height == 0 {
```

Comment:
Why is this the case? Height `0` corresponds to genesis block?

Reply:
Why is it not allowed to get a genesis block?

Reply:
Because we only store genesis header, and not whole block.

Reply:
Maybe term "genesis" is not correct. MockDa does not have any actual genesis, but we want to have finalized header and head header immediately as it starts, otherwise we have chicken and egg problem when starting.